modules/shared/src/main/scala/org/tessellation/schema/Fiber.scala

```diff
@@ -1,6 +1,6 @@
 package org.tessellation.schema
 
 trait Fiber[A, B] {
-  def parent: A
-  def reference: B
+  def reference: A
+  def data: B
```

Comment:
Originally in my understanding the Fiber was supposed to be an abstraction allowing for the objects of the same type that implement `Fiber` to be interconnected. Hence we had `parent` referencing previous tx/blocks/snapshot/whatever, `reference` being something that this object can be referenced by (so `hash` for `block`, `hash + ordinal` for `transaction`) which is used as `parent` for child objects. Now the `data` itself isn't something that can be referenced directly. Is that okay and as intended? I guess if we had a function calculating `parent` from `data` implemented for every `Fiber` that would make it work, though because of the hashing it wouldn't be pure and would need to be evaluated for every object being determined as parent. I'm not sure though what is the intended use of `Fiber`, so just posting my thoughts and how I understood it some time ago. Maybe you could share your reflections on my thoughts @buckysballs?

Reply:
Yep it's indeed the case that Fiber is meant to provide covariance for linked structures (single like Transaction, multiple/graph like block). This will help with validation and ledger updates for custom data types for state channels. We should be able to access the polymorphic attributes as long as they are specified. But this will allow for the acceptance and update pipelines to be generic, which we will want for when we expand to new types of Cells